nx/lib/nx/shared.ex

```diff
@@ -290,8 +290,8 @@ defmodule Nx.Shared do
       defmacro unquote(name)(unquote_splicing(args)) do
         {module, name} =
           case __CALLER__.context do
-            :guard -> {Kernel, unquote(name)}
-            _ -> {__MODULE__, unquote(fallback)}
+            ctx when ctx not in [:guard, :match] -> {__MODULE__, unquote(fallback)}
```

Comment:
As I'm not sure which values are valid, I chose to explicitly allow :guard and :match only

Reply:
it is only guard/match/nil.